libcni/api.go

```diff
@@ -271,6 +276,9 @@ func (c *CNIConfig) getCachedConfig(netName string, rt *RuntimeConf) ([]byte, *R
 		newRt.Args = unmarshaled.CniArgs
 	}
 	newRt.CapabilityArgs = unmarshaled.CapabilityArgs
+	if unmarshaled.IfName != "" {
+		newRt.IfName = unmarshaled.IfName
+	}
```

Comment:
How about adding some checks for ContainerID and NetworkName here?

Reply:
@mars1024 I should actually remove this line because if the interface name isn't already in newRt (because it was copied from the incoming RuntimeConf) then we couldn't have found the cache file since ContainerID and IfName are part of the filename generated by getCacheFilePath()

Reply:
removed and a testcase added

Reply:
emmmm... Good catch! lgtm except a small question, since filename includes ContainerID, IfName and NetworkName, why do we need to cache them in file content? Looks like they are not used anywhere. @dcbw